include/deal.II/lac/petsc_ts.templates.h

```diff
@@ -853,21 +1112,50 @@ namespace PETScWrappers
         AssertPETSc(VecDestroy(&av));
         AssertPETSc(VecDestroy(&rv));
       }
+  }
+
+
+
+  template <typename VectorType, typename PMatrixType, typename AMatrixType>
+  DEAL_II_CXX20_REQUIRES(
+    (concepts::is_dealii_petsc_vector_type<VectorType> ||
+     std::constructible_from<
+       VectorType,
+       Vec>)&&(concepts::is_dealii_petsc_matrix_type<PMatrixType> ||
+               std::constructible_from<
+                 PMatrixType,
+                 Mat>)&&(concepts::is_dealii_petsc_matrix_type<AMatrixType> ||
+                         std::constructible_from<AMatrixType, Mat>))
+  unsigned int TimeStepper<VectorType, PMatrixType, AMatrixType>::solve(
+    VectorType &y)
+  {
+    try
+      {
+        setup_callbacks();
+      }
+    catch (...)
+      {
+        throw;
+      }
```

Comment:
Same comment as above: This is not the place to catch user exceptions.

Separately, these lines are completely equivalent to simply saying
```suggestion
      setup_callbacks();
```
:-)

Reply:
Thanks, I'm not a C++ expert. Will fix

Reply:
done